### PR TITLE
Adding Usecase Documentation

### LINKED ISSUE #579: I want user-specific documentation on what an AaC usecase is, represents, and can be used for
As an AaC user, I want user-specific documentation on what an AaC usecase is, represents, and can be used for, so that I can get a more intimate familiarity with the `usecase` root type.

## AC:
- [x] There is user documentation for the AaC `usecase` concept
- [x] The `usecase` documentation explains:
  - [x] What kind of data does the `usecase` definition represent?
  - [x] What kind of data is represented in `usecase` definition sub-structures such as `steps`
  - [x] How would a user incorporate `usecase` into a modeled system (e.g. using usecase to represent behaviors and event flows) 

## Reminder
Reach out to JD with any questions or at a bare minimum include on PR to be sure we have made correct assumptions and explanations for usecases within AaC.

### PR DESCRIPTION
**Pull request by lizzcondrey**
# Description
Please include a summary of the changes and the related issue. List any dependencies that are required for this change.

# Linked Items:
Please include the items resolved by and/or associated with this Pull Request using the following syntax to link them for closure upon completion.
Closes #579 

## Type of change
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Feature improvement

### Feature Additions
Include pertinent changes as a result of adding the new feature.
- Adding documentation for `usecase` within AaC
- Modified TOC for User Guide to have new grouping for definition types
- Tweaked some code snippet highlighting that Sphinx linting flagged

### Bug Fixes
Include pertinent changes of fixing the bug.
- None

### Breaking Changes
Include pertinent changes that qualify as breaking changes. Be sure to include information about the impact to end users and additional information for addressing the break if more than a version transition is necessary for mitigating.
- None

### Feature Improvement Changes
Include pertinent changes that are improvements to existing features.
- None

# Checklist:
- [x] Updated documentation accordingly.
- [x] My changes generate no new warnings.
- [x] Have updated new and existing unit tests accordingly.
- [x] Linked associated items to be closed.
- [x] Have you bumped the version if there will be a release?

### CODE REVIEW

**Comment by lizzcondrey** [comment]
@jondavid-black we had this story flagged as wanting to be sure you were involved in the PR to make sure we had the correct understanding for the write up about usecases.

File: .github/workflows/main-branch.yml [hunk 1]
```diff
@@ -24,6 +24,9 @@ jobs:
     name: Python Cross Platform Build & Test
     uses: ./.github/workflows/python-cross-platform-support.yml
 
-  build_and_deploy_documentation:
-    name: Build and Deploy GitHub Pages
-    uses: ./.github/workflows/sphinx.yml
+# Commenting out from main branch workflow till permissions aspect can be resolved for
```
Comment (lizzcondrey) [inline .github/workflows/main-branch.yml:27]: Believe my added comment explains the drive for this change. Also, will let us evaluate if anything else within the main workflow has been impacted by the repo change and permissions needing updated.

Comment (Coffee2Bits) [reply .github/workflows/main-branch.yml:27]: I'm not tracking what the underlying issue for the page permission/token issue, are you familiar with that issue or is this something we need to invest some time into figuring out?

File: docs/source/project_documentation/user_guide/definition_types.rst [hunk 2]
```diff
@@ -0,0 +1,13 @@
+Definition Types in AaC
+=======================
+
+There are various types of definitions utilized in AaC to provide various system components. They are each represnted by a designated root key.
```
Comment (Coffee2Bits) [inline docs/source/project_documentation/user_guide/definition_types.rst:4]:
```suggestion
There are various types of definitions utilized in AaC to define system components. They are each represented by a distinct root key.
```

File: docs/source/project_documentation/user_guide/usecase_definitions.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
+# Usecase Definitions in AaC
+
+One of AaC's goal is to allow for a system to be modeled in a way that various disciplined users would be able to understand and implement the model for their contribution to the system. One of the system model aspects captured within AaC is the behavior of the system, which has a rootkey of `usecase`. The `usecase` describes a system behavior, who is involved with the system behavior, and outlines the steps for achieving the described system behavior.
```
Comment (Coffee2Bits) [inline docs/source/project_documentation/user_guide/usecase_definitions.md:3]:
```suggestion
One of AaC's goals is to allow for a system to be modeled in a way that various disciplined users are able to understand and implement the model for their area of responsibility in the overall system. One of the system model aspects captured within AaC is the behavior of the system, which has a root key of `usecase`. The `usecase` describes a system behavior, who is involved with the system behavior, and outlines the steps for achieving the described system behavior.
```

File: docs/source/project_documentation/user_guide/usecase_definitions.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+There is an example usecase within our alarm clock model example for setting the alarm time.
+
+```{eval-rst}
+.. literalinclude:: ../../../../python/model/alarm_clock/usecase.yaml
+    :language: yaml
+    :lines: 6-24
+```
+By incorporating the `usecase`s for various system behaviors, the alarm clock is able to be evaluated for functional completeness that cannot be seen in the alarm clock model alone.
+
+Some questions that can when evaluating the system for functional completeness are below. They can be addressed with the incorporation of a `usecase` definition for the system.
```
Comment (Coffee2Bits) [inline docs/source/project_documentation/user_guide/usecase_definitions.md:16]:
```suggestion
Some questions that come up when evaluating the system for functional completeness are outlined below. They can be addressed with the incorporation of a `usecase` definition for the system.
```

File: docs/source/project_documentation/user_guide/usecase_definitions.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+1. Does the system allow for necessary participants?
```
Comment (Coffee2Bits) [inline docs/source/project_documentation/user_guide/usecase_definitions.md:18]: Might just be a limit of my understanding, but allowing for a model seems like an ambiguous alternate of 'define'.
```suggestion
1. Does the system define the necessary participants?
```

File: docs/source/project_documentation/user_guide/usecase_definitions.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+The `steps` of the `usecase` are included as a portion of the `usecase` definition, highlighted below.
+
+```{eval-rst}
+.. literalinclude:: ../../../../python/model/alarm_clock/usecase.yaml
+    :language: yaml
+    :emphasize-lines: 11-19
+    :lines: 6-24
+```
+
+`steps` are actions that the system follows to accomplish the desired described behavior of the `usecase`. The number of steps varies based on how many interactions between the `participants` need to happen for the behavior to be accomplished. Each of the `steps` should involve only two `participants`. One participant (`source`) performing an `action` on another participant (`target`).
```
Comment (Coffee2Bits) [inline docs/source/project_documentation/user_guide/usecase_definitions.md:52]: I added a couple lines about making sure that these actions are defined by behaviors in the participants, however, we may need to discuss this further. I thought that JD had said that the actions should be defined in the source, but the alarm clock examples seem to lend themselves more to the actions being defined in the target. This suggestion can be ignored/set aside for a future story if we need to define this better amongst the team.
```suggestion
`steps` are actions that the system follows to accomplish the desired described behavior of the `usecase`. The number of steps varies based on how many interactions between the `participants` need to happen for the behavior to be accomplished. Each of the `steps` should involve only two `participants`. One participant (`source`) performing an `action` on another participant (`target`). Each action must be defined as a behavior in the `target`. For example, the model `AlarmClock` must have a behavior named `setAlarm` that can be invoked by the source `user`.
```

**Review by Coffee2Bits — CHANGES_REQUESTED** [review]
Looks good! There are a few editorial suggestions. I'll approve after they've been added; I'll keep an eye on my email so that I can promptly approve after you make the changes.

File: .github/workflows/main-branch.yml (hunk 1 above)
Comment (lizzcondrey) [reply .github/workflows/main-branch.yml:27]: My initial assessment thinks it's just likely to be the pages sourcing might not have been properly configured. But I can't confirm without access to settings for the repo. 
https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site

Seems to be problematic for automated publishing, but manual "bypasses".

File: docs/source/project_documentation/user_guide/usecase_definitions.md (hunk 6 above)
Comment (lizzcondrey) [reply docs/source/project_documentation/user_guide/usecase_definitions.md:52]: @jondavid-black have comments/clarification about current usecase components and where the behaviors are best defined?

File: .github/workflows/main-branch.yml (hunk 1 above)
Comment (Coffee2Bits) [reply .github/workflows/main-branch.yml:27]: I see. That's so odd...

Comment (lizzcondrey) [reply .github/workflows/main-branch.yml:27]: Yeah, the write ups on pages and their sources are a bit confusing on that portion. Single contributions (which I think of as manual) can be contributed to in varied ways with seeming no issue. But routine contributions (so like via the pipeline), has to have repo level configuration in place correctly.

**Review by Coffee2Bits — APPROVED** [review]

**Pull request merged**